module/mempool/consensus/exec_fork_suppressor.go

```diff
@@ -171,15 +147,45 @@ func (s *ExecForkSuppressor) Add(newSeal *flow.IncorporatedResultSeal) (bool, er
 // All returns all the IncorporatedResultSeals in the mempool
 func (s *ExecForkSuppressor) All() []*flow.IncorporatedResultSeal {
 	s.mutex.RLock()
-	defer s.mutex.RUnlock()
-	return s.seals.All()
+	seals := s.seals.All()
+	s.mutex.RUnlock()
+	sealsByBlockID := make(map[flow.Identifier]sealsList, 0)
+	for _, seal := range seals {
+		sealsPerBlock := sealsByBlockID[seal.Seal.BlockID]
+		sealsByBlockID[seal.Seal.BlockID] = append(sealsPerBlock, seal)
+	}
+
+	return s.filterConflictingSeals(sealsByBlockID)
 }
 
 // ByID returns an IncorporatedResultSeal by its ID
```

Comment:
```suggestion
// ByID returns an IncorporatedResultSeal by its ID
// the IncorporatedResultSeal's ID is the same as IncorporatedResult's ID, 
// so this call essentially is to find the seal for the incorporated result in the mempool. 
// Note: This call might crash if the block of the seal has multiple seals in mempool for conflicting 
// incorporated results. Usually the builder will call this method to find a seal for an incorporated 
// result, so the builder might crash if multiple conflicting seals exist.
```

We'd better add similar comment to the builder where ByID is used